cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -796,6 +798,242 @@ var _ = Describe("Operation Controller", func() {
 			)
 
 		})
+
+		Context("Process SyncOperation", func() {
+
+			var (
+				applicationDB          *db.Application
+				createOperationDBAndCR func(resourceID string)
+			)
+
+			BeforeEach(func() {
+				_, managedEnvironment, _, _, _, err := db.CreateSampleData(dbQueries)
+				Expect(err).To(BeNil())
+
+				_, dummyApplicationSpecString, err := createDummyApplicationData()
+				Expect(err).To(BeNil())
+
+				gitopsEngineCluster, _, err := dbutil.GetOrCreateGitopsEngineClusterByKubeSystemNamespaceUID(ctx, string(kubesystemNamespace.UID), dbQueries, logger)
+				Expect(gitopsEngineCluster).ToNot(BeNil())
+				Expect(err).To(BeNil())
+
+				By("creating a gitops engine instance with a namespace name/uid that don't exist in fakeclient")
+				gitopsEngineInstance := &db.GitopsEngineInstance{
+					Gitopsengineinstance_id: "test-fake-engine-instance",
+					Namespace_name:          workspace.Namespace,
+					Namespace_uid:           string(workspace.UID),
+					EngineCluster_id:        gitopsEngineCluster.Gitopsenginecluster_id,
+				}
+				err = dbQueries.CreateGitopsEngineInstance(ctx, gitopsEngineInstance)
+				Expect(err).To(BeNil())
+
+				applicationDB = &db.Application{
+					Application_id:          "test-my-application",
+					Name:                    name,
+					Spec_field:              dummyApplicationSpecString,
+					Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+					Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+				}
+
+				By("create Application in Database")
+				err = dbQueries.CreateApplication(ctx, applicationDB)
+				Expect(err).To(BeNil())
+
+				createOperationDBAndCR = func(resourceID string) {
```

Comment:
Think we can define the function before the `BeforeEach`, instead e.g.:

```go


		Context("Process SyncOperation", func() {

			createOperationDBAndCR := func( /* ... */) { /* function contents * }

			var (
				applicationDB          *db.Application
			)

```

In practice it doesn't make much difference, but the code is a tad more clear